lib/marginalia/comment.rb

```diff
@@ -26,9 +26,18 @@ def self.construct_comment
         end
       end
       ret.chop!
+      ret = self.escape_sql_comment(ret)
       ret
     end
 
+    def self.escape_sql_comment(str)
+      str = str.dup
```

Comment:
Can't we only `dup` if the `str` includes the patterns bellow? I want to save the extra allocation if not necessary.